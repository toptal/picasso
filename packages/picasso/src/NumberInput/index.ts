import type { OmitInternalProps } from '@toptal/picasso-shared'

import type { Props } from './NumberInput'

export { default } from './NumberInput'
export type NumberInputProps = OmitInternalProps<Props>
export * from './NumberInput'
