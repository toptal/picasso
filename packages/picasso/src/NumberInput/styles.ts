import { createStyles } from '@material-ui/core/styles'

export default () =>
  createStyles({
    root: {
      paddingRight: 0,
      cursor: 'text',
    },
    input: {
      '&::-webkit-outer-spin-button, &::-webkit-inner-spin-button': {
        '-webkit-appearance': 'none',
        appearance: 'none',
        margin: 0,
      },
      '-moz-appearance': 'textfield',
    },
  })
