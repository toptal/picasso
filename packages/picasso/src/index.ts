export { BaseProps } from '@toptal/picasso-shared'
export type { ColorType, SpacingType, SizeType } from '@toptal/picasso-shared'
export { AccordionCompound as Accordion } from './AccordionCompound'
export type { AccordionProps } from './Accordion'
export { default as AccountSelect } from './AccountSelect'
export type { AccountSelectProps } from './AccountSelect'
export { default as Amount } from './Amount'
export type { AmountProps } from './Amount'
export { default as Autocomplete } from './Autocomplete'
export type {
  AutocompleteProps,
  Item as AutocompleteItem,
} from './Autocomplete'
export { AvatarCompound as Avatar } from './AvatarCompound'
export type { AvatarProps } from './Avatar'
export { ButtonCompound as Button } from './ButtonCompound'
export type { ButtonProps, VariantType as ButtonVariantType } from './Button'
export type { ButtonCheckboxProps } from './ButtonCheckbox'
export type { ButtonRadioProps } from './ButtonRadio'
export { CheckboxCompound as Checkbox } from './CheckboxCompound'
export type { CheckboxProps } from './Checkbox'
export type { CheckboxGroupProps } from './CheckboxGroup'
export { default as Container } from './Container'
export type { ContainerProps } from './Container'
export { DropdownCompound as Dropdown } from './DropdownCompound'
export type { DropdownProps } from './Dropdown'
export { default as DropdownArrow } from './DropdownArrow'
export type { DropdownArrowProps } from './DropdownArrow'
export { default as EnvironmentBanner } from './EnvironmentBanner'
export type { EnvironmentBannerProps } from './EnvironmentBanner'
export { default as FileInput } from './FileInput'
export type { FileInputProps } from './FileInput'
export { default as FileList } from './FileList'
export type { FileListProps } from './FileList'
export { FormCompound as Form } from './FormCompound'
export type { FormProps } from './Form'
export { default as FormHint } from './FormHint'
export type { FormHintProps } from './FormHint'
export { default as FormError } from './FormError'
export type { FormErrorProps } from './FormError'
export { default as FormLabel } from './FormLabel'
export type { FormLabelProps, RequiredDecoration } from './FormLabel'
export { GridCompound as Grid } from './GridCompound'
export type { GridProps, GridSize } from './Grid'
export { HelpboxCompound as Helpbox } from './HelpboxCompound'
export type { HelpboxProps } from './Helpbox'
export { default as Image } from './Image'
export type { ImageProps } from './Image'
export { default as Indicator } from './Indicator'
export type { IndicatorProps } from './Indicator'
export { default as Input } from './Input'
export type { InputProps } from './Input'
export { default as InputAdornment } from './InputAdornment'
export type { InputAdornmentProps } from './InputAdornment'
export { TagCompound as Tag } from './TagCompound'
export type { TagProps } from './Tag'
export type { TagRectangularProps } from './TagRectangular'
export { default as Link } from './Link'
export type { LinkProps } from './Link'
export { default as Loader } from './Loader'
export type { LoaderProps } from './Loader'
export { default as Logo } from './Logo'
export type { LogoProps } from './Logo'
export { MenuCompound as Menu } from './MenuCompound'
export type { MenuProps } from './Menu'
export { default as MenuItem } from './MenuItem'
export type { MenuItemProps } from './MenuItem'
export { ModalCompound as Modal } from './ModalCompound'
export type { ModalProps } from './Modal'
export { default as MonthSelect } from './MonthSelect'
export type { MonthSelectProps } from './MonthSelect'
export { NotificationCompound as Notification } from './NotificationCompound'
export type { NotificationProps } from './Notification'
export { default as NotificationActions } from './NotificationActions'
export type { NotificationActionsProps } from './NotificationActions'
export { default as NumberInput } from './NumberInput'
export type { NumberInputProps } from './NumberInput'
export { PageCompound as Page } from './PageCompound'
export type { PageProps } from './Page'
export { default as Pagination } from './Pagination'
export type { PaginationProps } from './Pagination'
export { default as Paper } from './Paper'
export type { PaperProps } from './Paper'
export { default as PromptModal } from './PromptModal'
export type { PromptModalProps } from './PromptModal'
export { RadioCompound as Radio } from './RadioCompound'
export type { RadioProps } from './Radio'
export type { RadioGroupProps } from './RadioGroup'
export { default as RichTextEditor } from './RichTextEditor'
export type {
  RichTextEditorProps,
  RichTextEditorChangeHandler,
} from './RichTextEditor'
export { default as Select } from './Select'
export type { SelectProps } from './Select'
export type {
  Option as SelectOption,
  ValueType as SelectValueType,
} from './SelectBase'
export type { PageSidebarProps } from './PageSidebar'
export { default as Slider } from './Slider'
export type { SliderProps } from './Slider'
export { default as ShowMore } from './ShowMore'
export type { ShowMoreProps } from './ShowMore'
export { StepperCompound as Stepper } from './StepperCompound'
export type { StepperProps } from './Stepper'
export type { StepperVerticalProps } from './StepperVertical'
export { TableCompound as Table } from './TableCompound'
export type { TableProps } from './Table'
export { default as TableFooter } from './TableFooter'
export type { TableFooterProps } from './TableFooter'
export { TabsCompound as Tabs } from './TabsCompound'
export type { TabsProps } from './Tabs'
export { TagSelectorCompound as TagSelector } from './TagSelectorCompound'
export type { TagSelectorProps } from './TagSelector'
export { default as Tooltip } from './Tooltip'
export type { TooltipProps } from './Tooltip'
export { default as Typography } from './Typography'
export type { TypographyProps } from './Typography'
export { default as UserBadge } from './UserBadge'
export type { UserBadgeProps } from './UserBadge'
export { default as YearSelect } from './YearSelect'
export type { YearSelectProps } from './YearSelect'
export { default as ApplicationUpdateNotification } from './ApplicationUpdateNotification'
export type { ApplicationUpdateNotificationProps } from './ApplicationUpdateNotification'
export { AlertCompound as Alert } from './AlertCompound'
export type { AlertProps } from './Alert'
export type { AlertInlineProps } from './AlertInline'
export { default as Switch } from './Switch'
export type { SwitchProps } from './Switch'
export { default as Quote } from './Quote'
export type { QuoteProps } from './Quote'
export { default as Rating } from './Rating'
export type { RatingStarsProps, RatingThumbsProps } from './Rating'
export {
  default as TreeView,
  TreeViewContainer,
  useTreeView,
  TreeNodeAvatar,
} from './TreeView'
export type { TreeNodeInterface } from './TreeView'
export { default as StaticTreeView } from './StaticTreeView'
export { default as TimePicker } from './TimePicker'
export type { TimePickerProps } from './TimePicker'
export { default as SkeletonLoader } from './SkeletonLoader'
export { default as PageHead } from './PageHead'
export type { PageHeadProps } from './PageHead'
export { default as ProgressBar } from './ProgressBar'
export type { ProgressBarProps } from './ProgressBar'
export { default as List } from './List'
export type { ListProps } from './List'
export { default as Badge } from './Badge'
export type { BadgeProps } from './Badge'
export { BreadcrumbsCompound as Breadcrumbs } from './BreadcrumbsCompound'
export type { BreadcrumbsProps } from './Breadcrumbs'
export { default as Calendar } from './Calendar'
export type { DateOrDateRangeType } from './Calendar'
export {
  default as DatePicker,
  datePickerParseDateString,
  DEFAULT_DATE_PICKER_DISPLAY_DATE_FORMAT,
  DEFAULT_DATE_PICKER_EDIT_DATE_FORMAT,
} from './DatePicker'
export type {
  DatePickerValue,
  DatePickerInputCustomValueParser,
  DatePickerInputCustomValueParserParameters,
  DatePickerProps,
} from './DatePicker'
export { default as Drawer } from './Drawer'
export type { DrawerProps } from './Drawer'
export { default as Dropzone, ErrorCode as DropzoneErrorCode } from './Dropzone'
export type {
  DropzoneProps,
  FileUpload as DropzoneFileUpload,
  FileError as DropzoneFileError,
  FileRejection as DropzoneFileRejection,
  DropEvent as DropzoneDropEvent,
} from './Dropzone'
export { OverviewBlockCompound as OverviewBlock } from './OverviewBlockCompound'
export type { OverviewBlockProps } from './OverviewBlock'
export { default as TypographyOverflow } from './TypographyOverflow'
export type { TypographyOverflowProps } from './TypographyOverflow'
export { default as EmptyState } from './EmptyState'
export { default as Section } from './Section'
export type { SectionProps } from './Section'
export { NoteCompound as Note } from './NoteCompound'
export type { NoteProps } from './Note'
export { default as Timeline } from './Timeline'
export type { TimelineProps } from './Timeline'
export { default as RichText } from './RichText'
export type { RichTextProps, ASTType } from './RichText'
export { default as PasswordInput } from './PasswordInput'
export type { PasswordInputProps } from './PasswordInput'
export { default as FieldRequirements } from './FieldRequirements'
export type {
  FieldRequirementsProps,
  FieldRequirement,
} from './FieldRequirements'
export type { Status as OutlinedInputStatus } from './OutlinedInput'
export { default as AvatarUpload } from './AvatarUpload'
export type {
  AvatarUploadProps,
  FileUpload as AvatarUploadFileUpload,
  FileError as AvatarUploadFileError,
  FileRejection as AvatarUploadFileRejection,
  DropEvent as AvatarUploadDropEvent,
} from './AvatarUpload'

// hygen code generator inserts export statements above this comment.
export * from './Icon'
