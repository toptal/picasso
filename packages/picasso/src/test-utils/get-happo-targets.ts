type Result = Record<string, any>

/**
 * Generates Happo targets for responsive visual tests
 *
 * @param screenWidths Screen widths to generate targets for
 * @returns {Result} Object with Happo targets
 */
export const getHappoTargets = (screenWidths: number[]) =>
  screenWidths.reduce<Result>((acc: Result, width: number) => {
    const name = `chrome-desktop-width-${width}`

    acc[name] = {
      name,
      browser: 'chrome',
      viewport: `${width}x1024`,
    }

    return acc
  }, {})
