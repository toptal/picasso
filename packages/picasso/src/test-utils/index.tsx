import type { ReactElement } from 'react'
import React from 'react'
import type { RenderOptions, RenderResult } from '@testing-library/react'
import { render } from '@testing-library/react'

import TestingPicasso from './TestingPicasso'

export type PicassoConfig = {
  titleCase: boolean
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>,
  picassoConfig?: PicassoConfig
): RenderResult => {
  const { rerender, ...rest } = render(
    <TestingPicasso titleCase={picassoConfig?.titleCase}>{ui}</TestingPicasso>,
    options
  )

  const customRerender: typeof rerender = newUi =>
    rerender(
      <TestingPicasso titleCase={picassoConfig?.titleCase}>
        {newUi}
      </TestingPicasso>
    )

  return { rerender: customRerender, ...rest }
}

export * from '@testing-library/react'
export { customRender as render, TestingPicasso }
export { getCheckpoints } from './get-checkpoints'
