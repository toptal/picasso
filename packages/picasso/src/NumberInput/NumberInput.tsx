import React, { forwardRef, useRef, RefObject, ReactNode } from 'react'
import { makeStyles, Theme } from '@material-ui/core/styles'
import { BaseProps, OmitInternalProps, SizeType } from '@toptal/picasso-shared'
import ButtonBase from '@material-ui/core/ButtonBase'
import cx from 'classnames'

import OutlinedInput, { Props as OutlinedInputProps } from '../OutlinedInput'
import InputAdornment from '../InputAdornment'
import Container from '../Container'
import { useCombinedRefs } from '../utils'
import { ArrowDownMinor16, ArrowUpMinor16 } from '../Icon'
import styles from './styles'
import isBrowser from '../utils/is-browser'

export interface Props
  extends Omit<
      OmitInternalProps<OutlinedInputProps>,
      'defaultValue' | 'type' | 'multiline' | 'rows'
    >,
    BaseProps {
  /** Value of the `input` element. */
  value?: string | number
  /** Minimum value for the `input` element */
  min?: number | string
  /** Maximum value for the `input` element */
  max?: number | string
  /** Next value of the `input` element will be calculated based on step */
  step?: number | string
  /** Should controls be hidden or not */
  hideControls?: boolean
  /** Specify icon which should be rendered inside NumberInput */
  icon?: ReactNode
  /** Indicates whether component is in error state */
  error?: boolean
  /** Indicates whether component is in disabled state */
  disabled?: boolean
  /** Callback invoked when `NumberInput` changes its state. */
  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void
}

type NumberAdornmentProps = {
  min: number | string
  max: number | string
  step: number | string
  value: Props['value']
  inputRef: RefObject<HTMLInputElement>
  classes: Record<string, string>
  disabled?: boolean
  size?: SizeType<'small' | 'medium' | 'large'>
}

// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
const nativeInputValueSetter = isBrowser()
  ? (Object.getOwnPropertyDescriptor(
      window.HTMLInputElement.prototype,
      'value'
    ) as PropertyDescriptor).set!
  : undefined

const NumberAdornment = (props: NumberAdornmentProps) => {
  const {
    step,
    min,
    max,
    value,
    inputRef,
    classes,
    disabled,
    size = 'medium'
  } = props

  const normalizedStep = Number(step)
  const normalizedValue = Number(value)
  const normalizedMin = Number(min)
  const normalizedMax = Number(max)

  const fireEvent = (nextValue: number) => {
    if (!nativeInputValueSetter) {
      return
    }

    const input = inputRef.current

    nativeInputValueSetter.call(input, nextValue)

    const event = new Event('input', {
      bubbles: true,
      cancelable: true
    })

    if (input) {
      input.dispatchEvent(event)
    }
  }

  const handleUpClick = () => {
    if (typeof value === 'undefined') {
      return
    }

    let nextValue = normalizedValue + normalizedStep

    if (nextValue <= max) {
      if (normalizedValue < normalizedMin + normalizedStep) {
        nextValue = normalizedMin + normalizedStep
      }

      fireEvent(nextValue)
    } else if (normalizedValue !== normalizedMax) {
      nextValue = normalizedMax
      fireEvent(nextValue)
    }
  }

  const handleDownClick = () => {
    if (typeof value === 'undefined') {
      return
    }

    let nextValue = normalizedValue - normalizedStep

    if (nextValue >= min) {
      if (normalizedValue > normalizedMax - normalizedStep) {
        nextValue = normalizedMax - normalizedStep
      }

      fireEvent(nextValue)
    } else if (normalizedValue !== normalizedMin) {
      nextValue = normalizedMin
      fireEvent(nextValue)
    }
  }

  return (
    <InputAdornment position='end'>
      <Container flex direction='column' inline>
        <ButtonBase
          disabled={disabled}
          classes={{
            root: cx(classes.control, classes[size]),
            disabled: classes.controlDisabled
          }}
          onClick={handleUpClick}
        >
          <ArrowUpMinor16 />
        </ButtonBase>
        <ButtonBase
          disabled={disabled}
          classes={{
            root: cx(classes.control, classes[size]),
            disabled: classes.controlDisabled
          }}
          onClick={handleDownClick}
        >
          <ArrowDownMinor16 />
        </ButtonBase>
      </Container>
    </InputAdornment>
  )
}

const useStyles = makeStyles<Theme>(styles, {
  name: 'PicassoNumberInput'
})

export const NumberInput = forwardRef<HTMLInputElement, Props>(
  function NumberInput(props, ref) {
    const {
      step = 1,
      min = -Infinity,
      max = Infinity,
      hideControls,
      value,
      onChange,
      disabled,
      error,
      onResetClick,
      enableReset,
      width,
      icon,
      size,
      ...rest
    } = props

    const classes = useStyles()

    const inputRef = useCombinedRefs<HTMLInputElement>(
      ref,
      useRef<HTMLInputElement>(null)
    )

    const endAdornment = hideControls ? null : (
      <NumberAdornment
        step={step}
        min={min}
        max={max}
        value={value}
        inputRef={inputRef}
        classes={classes}
        disabled={disabled}
        size={size}
      />
    )

    const startAdornment = icon ? (
      <InputAdornment position='start' disablePointerEvents>
        {icon}
      </InputAdornment>
    ) : null

    return (
      <OutlinedInput
        classes={{
          root: classes.root,
          input: classes.input
        }}
        inputProps={{
          ...rest,
          step,
          min,
          max
        }}
        width={width}
        onResetClick={onResetClick}
        enableReset={enableReset}
        error={error}
        inputRef={inputRef}
        type='number'
        value={value}
        disabled={disabled}
        onChange={onChange}
        endAdornment={endAdornment}
        startAdornment={startAdornment}
        size={size}
      />
    )
  }
)

NumberInput.defaultProps = {
  onChange: () => {},
  value: 0,
  step: 1,
  min: -Infinity,
  max: Infinity,
  hideControls: false,
  size: 'medium'
}

NumberInput.displayName = 'NumberInput'

export default NumberInput
