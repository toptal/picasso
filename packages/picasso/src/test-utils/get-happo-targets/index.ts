import { getCheckpoints } from './get-checkpoints'
import { getHappoTargets } from './get-happo-targets'

const checkpoints = getCheckpoints()

export { getHappoTargets }

export const HAPPO_TARGETS = getHappoTargets(checkpoints)
