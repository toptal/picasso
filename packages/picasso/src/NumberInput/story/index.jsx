import { NumberInput } from '../NumberInput'
import PicassoBook from '~/.storybook/components/PicassoBook'

const page = PicassoBook.section('Forms').createPage(
  'NumberInput',
  'Input component for numbers'
)

page
  .createTabChapter('Props')
  .addComponentDocs({ component: NumberInput, name: 'NumberInput' })

page
  .createChapter()
  .addExample('NumberInput/story/Default.example.tsx', 'Default')
  .addExample('NumberInput/story/Disabled.example.tsx', 'Disabled')
  .addExample('NumberInput/story/Status.example.tsx', {
    title: 'Status',
    takeScreenshot: false
  })
  .addExample('NumberInput/story/WithIcon.example.tsx', 'With Icon')
  .addExample('NumberInput/story/Sizes.example.tsx', 'Sizes')
