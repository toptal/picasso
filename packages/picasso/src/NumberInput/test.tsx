import React, { useState, ChangeEventHandler } from 'react'
import { render, fireEvent } from '@toptal/picasso/test-utils'

import NumberInput, { Props as NumberInputProps } from './NumberInput'

const NumberInputRenderer = (
  props: { initialValue: number | string } & Partial<NumberInputProps>
) => {
  const [value, setValue] = useState(props.initialValue)

  const handleChange: ChangeEventHandler<HTMLInputElement> = e => {
    setValue(e.target.value)
  }

  return (
    <NumberInput
      step={5}
      max={100}
      min={-100}
      value={value}
      onChange={handleChange}
      status={props.status}
      testIds={props.testIds}
    />
  )
}

const renderNumberInput = (
  props: Partial<NumberInputProps> & { initialValue?: number | string } = {}
) => {
  const { initialValue = '10' } = props

  return render(<NumberInputRenderer initialValue={initialValue} {...props} />)
}

describe('NumberInput', () => {
  it('renders', () => {
    const { container } = renderNumberInput()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('increase value', async () => {
    const { getByDisplayValue, queryAllByRole } = renderNumberInput()

    const input = getByDisplayValue('10') as HTMLInputElement

    const controls = queryAllByRole('button')
    const controlUp = controls[0]

    fireEvent.click(controlUp)

    expect(input.value).toBe('15')
  })

  it('decrease value', () => {
    const { getByDisplayValue, queryAllByRole } = renderNumberInput()

    const input = getByDisplayValue('10') as HTMLInputElement

    const controls = queryAllByRole('button')
    const controlDown = controls[1]

    fireEvent.click(controlDown)

    expect(input.value).toBe('5')
  })

  describe('near max/min limits', () => {
    it('increase value near max limit', () => {
      const { getByDisplayValue, queryAllByRole } = renderNumberInput({
        initialValue: '97'
      })

      const input = getByDisplayValue('97') as HTMLInputElement

      const controls = queryAllByRole('button')
      const controlUp = controls[0]

      fireEvent.click(controlUp)

      expect(input.value).toBe('100')
    })

    it('decrease value near min limit', () => {
      const { getByDisplayValue, queryAllByRole } = renderNumberInput({
        initialValue: '-97'
      })

      const input = getByDisplayValue('-97') as HTMLInputElement

      const controls = queryAllByRole('button')
      const controlDown = controls[1]

      fireEvent.click(controlDown)

      expect(input.value).toBe('-100')
    })

    it('decrease value when it is closer than step to max', () => {
      const { getByDisplayValue, queryAllByRole } = renderNumberInput({
        initialValue: '97'
      })

      const input = getByDisplayValue('97') as HTMLInputElement

      const controls = queryAllByRole('button')
      const controlDown = controls[1]

      fireEvent.click(controlDown)

      expect(input.value).toBe('95')
    })

    it('increase value when it is closer to min than step', () => {
      const { getByDisplayValue, queryAllByRole } = renderNumberInput({
        initialValue: '-97'
      })

      const input = getByDisplayValue('-97') as HTMLInputElement

      const controls = queryAllByRole('button')
      const controlUp = controls[0]

      fireEvent.click(controlUp)

      expect(input.value).toBe('-95')
    })
  })

  describe('when in a valid state', () => {
    it('shows valid icon', () => {
      const testProps: NumberInputProps = {
        value: '10',
        status: 'success',
        testIds: { validIcon: 'valid-icon' }
      }

      const { getByTestId, rerender } = renderNumberInput(testProps)

      const validIcon = getByTestId('valid-icon')

      expect(validIcon).toBeVisible()

      // re-render with different props
      rerender(<NumberInput {...testProps} status='error' />)

      expect(validIcon).not.toBeVisible()
    })
  })
})
