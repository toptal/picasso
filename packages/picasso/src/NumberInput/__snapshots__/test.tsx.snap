// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberInput renders 1`] = `
<div
  class="Picasso-root"
>
  <div
    class="MuiInputBase-root MuiOutlinedInput-root PicassoOutlinedInput-root PicassoNumberInput-root PicassoOutlinedInput-rootAuto PicassoOutlinedInput-rootMedium MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
  >
    <input
      class="MuiInputBase-input MuiOutlinedInput-input PicassoOutlinedInput-input PicassoNumberInput-input PicassoOutlinedInput-inputMedium MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
      max="100"
      min="-100"
      step="5"
      type="number"
      value="10"
    />
    <div
      class="MuiInputAdornment-root PicassoInputAdornment-root MuiInputAdornment-positionEnd"
    >
      <div
        class="PicassoContainer-flex PicassoContainer-inline PicassoContainer-column"
      >
        <button
          class="MuiButtonBase-root PicassoNumberInput-control PicassoNumberInput-medium"
          tabindex="0"
          type="button"
        >
          <svg
            class="PicassoSvgArrowUpMinor16-root"
            style="min-width: 16px; min-height: 16px;"
            viewBox="0 0 16 16"
          >
            <path
              d="M7.997 5.29l4.707 4.707-.707.707-4-4-4 4-.707-.707 4-4 .707-.707z"
            />
          </svg>
        </button>
        <button
          class="MuiButtonBase-root PicassoNumberInput-control PicassoNumberInput-medium"
          tabindex="0"
          type="button"
        >
          <svg
            class="PicassoSvgArrowDownMinor16-root"
            style="min-width: 16px; min-height: 16px;"
            viewBox="0 0 16 16"
          >
            <path
              d="M11.997 5.29l.707.707-4 4-.707.707-.707-.707-4-4 .707-.707 4 4 4-4z"
            />
          </svg>
        </button>
      </div>
    </div>
    <fieldset
      aria-hidden="true"
      class="PrivateNotchedOutline-root MuiOutlinedInput-notchedOutline PicassoOutlinedInput-notchedOutline"
      style="padding-left: 8px;"
    >
      <legend
        class="PrivateNotchedOutline-legend"
        style="width: 0.01px;"
      >
        <span>
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
